python/pyiceberg/catalog/hive.py

```diff
@@ -201,17 +238,46 @@ def __init__(self, name: str, properties: Properties, uri: str):
         super().__init__(name, properties)
         self._client = _HiveClient(uri)
 
-    def _convert_hive_into_iceberg(self, table: HiveTable) -> Table:
-        # Requires reading the manifest, will implement this in another PR
-        # Check the table type
-        return Table(identifier=(table.dbName, table.tableName))
+    def _convert_hive_into_iceberg(self, table: HiveTable, io: FileIO, metadata_location: Optional[str] = None) -> Table:
+        properties: Dict[str, str] = table.parameters
+        if TABLE_TYPE not in properties:
+            raise NoSuchTableError(f"Property table_type missing, could not determine type: {table.dbName}.{table.tableName}")
+
+        table_type = properties[TABLE_TYPE]
+        if table_type.lower() != ICEBERG:
+            raise NoSuchTableError(f"Property table_type is {table_type}, expected {ICEBERG}: {table.dbName}.{table.tableName}")
+
+        if not metadata_location:
+            if prop_metadata_location := properties.get(METADATA_LOCATION):
+                metadata_location = prop_metadata_location
+            else:
+                raise NoSuchTableError("Metadata location not found")
+
+        file = io.new_input(metadata_location)
+        metadata = FromInputFile.table_metadata(file)
+        return Table(identifier=(table.dbName, table.tableName), metadata=metadata, metadata_location=metadata_location)
+
+    def _write_metadata(self, metadata: TableMetadataV2, io: FileIO, metadata_path: str):
```

Comment:
Why does this accept metadata v2 instead of either v1 or v2? I don't think that we want to assume that we will only write v2 metadata. We can't upgrade tables automatically.

Reply:
I've updated the signature. This is because we don't have any logic that allows updating a table. This is only used for creating new tables, and I think it makes sense to only allow V2 tables (or at least push the user into that direction).